### PR TITLE
Added "Migrate a React client-side application to server-side with Next.JS" post

### PR DESCRIPTION
**Pull request by eko**

### CODE REVIEW

File: _posts/2017-09-03-migrer-une-application-react-client-side-en-server-side-avec-nextjs.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    - Frontend
+tags:
+    - javascript
+    - server
+    - react
+    - nextjs
+cover: /assets/2017-09-03-migrer-une-application-react-client-side-en-server-side-avec-nextjs/cover.jpg
+---
+
+La plupart des applications front utilisant React sur lesquelles j'ai pu travaillé sont des applications destinées à être rendues par le navigateur (client-side).
```
Comment (GusTTShowbiz) [inline _posts/2017-09-03-migrer-une-application-react-client-side-en-server-side-avec-nextjs.md]: `j'ai pu travailler`

File: _posts/2017-09-03-migrer-une-application-react-client-side-en-server-side-avec-nextjs.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+Premiers pas
+------------
+
+Avant tout, il nous faut regarder ce que propose Next.JS afin que voir si ce framework correspond à nos besoins :
+
+* Une [installation simple](https://github.com/zeit/next.js#setup){:target="_blank"},
+* Rendu de [fichiers statiques](https://github.com/zeit/next.js#static-file-serving-eg-images){:target="_blank"} tel que des images et fichiers CSS dans un répertoire `static`,
+* Des URLs (routing) pré-définies à partir des noms des fichiers JS présents dans le répertoire `pages`,
+* Possibilité de définir des [routes personnalisées/paramétrisées](https://github.com/zeit/next.js/#custom-server-and-routing){:target="_blank"} (but I used [next-routes](https://github.com/fridays/next-routes){:target="_blank"} for this, a cleaner way),
```
Comment (GusTTShowbiz) [inline _posts/2017-09-03-migrer-une-application-react-client-side-en-server-side-avec-nextjs.md]: `but I used...` => in french please 🇫🇷

File: _posts/2017-09-03-migrer-une-application-react-client-side-en-server-side-avec-nextjs.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+Très simple, nous instancions un objet `Page` React comprenant les composants qui vont former notre page et appelons une fonction retournée par `withRedux(store)` (fourni par la librairie `next-redux-wrapper`) permettant de synchroniser notre store Redux sur cette page.
+
+Ainsi, toute notre page (et les composants compris à l'intérieur) auront accès et pourront manipuler le store Redux.
+
+Si vous vous rendez maintenant sur `http://localhost:3000/`, votre homepage doit maintenant être visible !
+
+Routes personnalisées / comprennant des variables
+-------------------------------------------------
+
+Malheureusement, il est un peu complexe de déclarer des routes personnalisées et/ou contenant des variables actuellement directement via Next.JS, c'est pourquoi j'ai fais le choix d'utiliser la librairie `next-routes` qui va me permettre de faire cela très simplement.
```
Comment (GusTTShowbiz) [inline _posts/2017-09-03-migrer-une-application-react-client-side-en-server-side-avec-nextjs.md]: `j'ai fait`

File: _posts/2017-09-03-migrer-une-application-react-client-side-en-server-side-avec-nextjs.md [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+Il nous faut alors déclarer un fichier à l'emplacement `src/routes.js` avec le contenu suivant :
+
+```js
+const routes = module.exports = require('next-routes')()
+
+routes
+  .add('category', '/category/:slug')
+```
+
+Lorsque l'URL `/category/:slug` sera appelée (ou slug est une valeur dynamique), le fichier `src/pages/category.js` sera appelé.
```
Comment (GusTTShowbiz) [inline _posts/2017-09-03-migrer-une-application-react-client-side-en-server-side-avec-nextjs.md]: `où`

**Comment by eko** [comment]
@GusTTShowbiz Thanks for your comments! Errors are fixed

**Pull request merged**